test.py

```diff
@@ -413,7 +418,7 @@ async def add_test(self, shortname, casename) -> None:
         args = self.custom_args.get(shortname, ["-c2 -m2G"])
         args = merge_cmdline_options(args, self.options.extra_scylla_cmdline_options)
         for a in args:
```

Comment:
This merge + iteration above is wrong, so 'test.py --extra-scylla-cmdline-options' is broken. Ideally, I'd have merged self.extra_cmdline_options (extra options retrived by yaml file instead) into args, but the bug made it impossible so I went with this approach instead. I am not willing to fix it now. Opened https://github.com/scylladb/scylladb/issues/22696 for us to track it.